src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int LENGHT = 10;
+    private Object[] array;
+    private int size;
+
+    public ArrayList() {
+        array = new Object[LENGHT];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        validateIndex(index, size + 1);
+        grow();
 
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
 
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+
     }
 
     @Override
     public T get(int index) {
-        return null;
+        validateIndex(index, size);
+        return (T) array[index];
     }
 
     @Override
     public void set(T value, int index) {
+        validateIndex(index, size);
+        array[index] = value;
 
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        validateIndex(index, size);
+        T removed = (T) array[index];
+        System.arraycopy(array, index + 1, array, index, size - index - 1);
+        size--;
+        return removed;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        T removed;
+        int index = getIndex(element);
+        if (index < 0) {
+            throw new NoSuchElementException("Element is not found in the array");
+        }
+        removed = (T) array[index];
+        remove(index);
+        return removed;
```

Comment:
```suggestion
        return  (T) array[index];
```
can delete staterment - removed

Reply:
i can not compile code with that suggestion

Reply:
I agree, my bad idea
But you can do something like this: 
```suggestion
                return remove(index);
```

Reply:
it is also will be mistake

Reply:
Your  suggestion  will be mistake, that code isn't executed